### PR TITLE
godot: update to 3.3.4

### PR DESCRIPTION
**Pull request by jasonliu--**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.11.6 15G22010 x86_64
Xcode 8.2.1 8C1002

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: games/godot/Portfile [hunk 1]
```diff
@@ -25,11 +25,11 @@ long_description    Godot Engine is a cross-platform game engine for \
 if {$subport eq ${name}} {
     if {${os.major} <= 15} { PortGroup legacysupport 1.1 }
```
Comment (reneeotten) [inline games/godot/Portfile:26]: this should ideally also check for the platform as in `{if ${os.platform} eq "darwin"` && .....`}

Comment (jasonliu--) [reply games/godot/Portfile:26]: I've always wondered about that.... Doesn't the `platforms darwin` line already imply that the entire file is for `${os.platform} eq "darwin"`? Or am I misunderstanding the purpose of `platforms darwin`?

(and by the way, I do use `${os.platform} eq "darwin"` most other places. I've just never understood why; I simply copied what I was seeing in other Portfiles)

Comment (reneeotten) [reply games/godot/Portfile:26]: please just add it, if you have it everywhere else in the Portfile and other ones use it as well (and it's always requested if not present when checking `os.major`). I am not 100% sure about it, but it doesn't hurt....

Comment (jasonliu--) [reply games/godot/Portfile]: I've gone ahead and made the requested change, but I also asked the question on the dev mailing list. It's interesting that the conversation seems to have brought up the possibility that wrapping chunks of code inside of a `platform darwin {` could eliminate the need to check for `${os.platform} eq "darwin"` each and every time that `${os.major}` is used.

**Review by reneeotten — APPROVED** [review]

**Pull request merged**